src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptions.cs

```diff
@@ -0,0 +1,328 @@
+        }
+
+        private static void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            Compilation compilation = context.Compilation;
+
+            compilation.TryGetOrCreateTypeByMetadataName(
+                WellKnownTypeNames.SystemTextJsonJsonSerializerOptions, out INamedTypeSymbol? jsonSerializerOptionsSymbol);
+
+            compilation.TryGetOrCreateTypeByMetadataName(
+                WellKnownTypeNames.SystemTextJsonJsonSerializer, out INamedTypeSymbol? jsonSerializerSymbol);
+
+            if (jsonSerializerOptionsSymbol is null || jsonSerializerSymbol is null)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                context =>
+                {
+                    var operation = (IObjectCreationOperation)context.Operation;
+
+                    INamedTypeSymbol? typeSymbol = operation.Constructor?.ContainingType;
+                    if (SymbolEqualityComparer.Default.Equals(typeSymbol, jsonSerializerOptionsSymbol))
+                    {
+                        if (IsCtorUsedAsArgumentForJsonSerializer(operation, jsonSerializerSymbol) ||
+                            IsLocalUsedAsArgumentForJsonSerializerOnly(operation, jsonSerializerSymbol))
+                        {
+                            context.ReportDiagnostic(operation.CreateDiagnostic(s_Rule));
+                        }
+                    }
+                },
+                OperationKind.ObjectCreation);
+        }
+
+        private static bool IsCtorUsedAsArgumentForJsonSerializer(IObjectCreationOperation objCreationOperation, INamedTypeSymbol jsonSerializerSymbol)
+        {
+            IOperation operation = WalkUpConditional(objCreationOperation);
+
+            return operation.Parent is IArgumentOperation argumentOperation &&
+                IsArgumentForJsonSerializer(argumentOperation, jsonSerializerSymbol);
+        }
+
+        private static bool IsArgumentForJsonSerializer(IArgumentOperation argumentOperation, INamedTypeSymbol jsonSerializerSymbol)
+        {
+            return argumentOperation.Parent is IInvocationOperation invocationOperation &&
+                SymbolEqualityComparer.Default.Equals(invocationOperation.TargetMethod.ContainingType, jsonSerializerSymbol);
+        }
+
+        private static bool IsLocalUsedAsArgumentForJsonSerializerOnly(IObjectCreationOperation objCreation, INamedTypeSymbol jsonSerializerSymbol)
+        {
+            IOperation operation = WalkUpConditional(objCreation);
+
+            if (!IsLocalAssignment(operation, out List<ILocalSymbol>? localSymbols))
+            {
+                return false;
+            }
+
+            IBlockOperation? localBlock = objCreation.GetFirstParentBlock();
+            bool ret = false;
+
+            foreach (ILocalReferenceOperation descendant in localBlock.Descendants().OfType<ILocalReferenceOperation>())
```

Comment:
Not sure how performance sensitive analyzers need to be, but you could avoid that Linq call with a check inside the loop:
 ```C#
if (descendant is not ILocalReferenceOperation localRefOperation &&
    !localSymbols.Contains(localRefOperation.Local))
{
    continue;
{
```